R/diff_secuTrial.R

```diff
@@ -0,0 +1,58 @@
+#' Returns differences in the setup of two secuTrialdata objects
+#' @description During ongoing studies it is possible that changes to the DataCapture interface
+#'              are made. Sometimes these changes may call for adjustments in analysis code.
+#'              It is considered good practice to run \code{diff_secuTrial()} on the last export
+#'              and the current export of a project to at least make yourself aware of
+#'              potential changes to the setup. If there are differences, the results of this function should
+#'              be interpreted as a first indicator since they may not cover all alterations.
+#'              Information is returned on new forms and variables.
+#'              A detailed list of changes can be produced in the FormBuilder with
+#'              "Compare project setup".
+#' @param x a \code{secuTrialdata} object (the older export)
+#' @param y a \code{secuTrialdata} object (the newer export)
+#' @export
+#' @details If there are differences, \code{diff_secuTrial()} will produce a list of vectors.
+#'          The fist vector informs about new forms and the second vector informs about
+#'          new variables.
+#'
+#' @examples
+#' # export location
+#' export_loc <- system.file("extdata", "sT_exports", "longnames",
+#'                           "s_export_CSV-xls_CTU05_long_ref_miss_en_utf8.zip",
+#'                           package = "secuTrialR")
+#' # read export
+#' sT_export <- read_secuTrial(export_loc)
+#'
+#' # return scores // in this case on the same export
+#' diff_secuTrial(sT_export, sT_export)
+#'
+diff_secuTrial <- function(x, y) {
+  if (class(x) == "secuTrialdata" & class(y) == "secuTrialdata") {
+
+    # comapring different projects makes no sense really
+    if(x$export_options$project_name != y$export_options$project_name) {
+      stop("The two exports appear to originate from different secuTrial projects.")
+    }
+
+    if(x$export_options$project_version == y$export_options$project_version) {
+      print(paste0("The project structure has not changed. Project versions are both: ",
+                   x$export_options$project_version))
+    } else if (x$export_options$project_version != y$export_options$project_version) {
+      # forms
+      forms_x <- x[[x$export_options$meta_names$forms]]
+      forms_y <- y[[y$export_options$meta_names$forms]]
+      form_diff <- unique(forms_y$formtablename[which(! forms_y$formtablename %in% forms_x$formtablename)])
```

Comment:
If you have two different versions that contain the same form tables, but one export is with "long", the other with "short" names - would all form table names light up as being different here, even if they are equivalent? Can we avoid this?

Reply:
Good point. I will check that and if it is as you suspect I will add a check for this export option.

Reply:
I'd still want to know... it makes a difference to the code I've written to analyze the data! Short names won't work if i've written my codes based on an export with long names...

Reply:
So you would like a check for export option equality?

Reply:
I think that the same names are there is probably OK. Personally, I wouldn't reference a given object by it's position in the list, but by it's name (so it's not important where in the list the object is). If the position in the list is needed, then equality would need to be checked, otherwise just that all forms in x are in y and vice versa I think

Reply:
The names are the same irrespective of long or short names. So I think we are good here?
```
> sdat_ctu05$export_options$short_names
[1] TRUE
> sdat_ctu05$fs$formtablename
[1] "mnpctu05baseline"      "emnpctu05surgeries"    "mnpctu05outcome"       "mnpctu05treatment"     "mnpctu05allmedi"      
[6] "mnpctu05studyterminat" "mnpctu05ae"            "mnpctu05sae"          
> ldat_ctu05$export_options$short_names
[1] FALSE
> ldat_ctu05$forms$formtablename
[1] "mnpctu05baseline"      "emnpctu05surgeries"    "mnpctu05outcome"       "mnpctu05treatment"     "mnpctu05allmedi"      
[6] "mnpctu05studyterminat" "mnpctu05ae"            "mnpctu05sae"
```

Reply:
Super. All good for me.